src/app/components/history/history.component.html

```diff
@@ -1,19 +1,36 @@
 <div class="container-fluid">
+  <div class="row mb-2">
+    <div class="btn-group btn-group-toggle">
+      <label class="btn btn-secondary" for="filter_all" [class.active]="filter === 'all'">
+        <input type="radio" [ngModel]="filter" value="all" name="options" id="filter_all" (ngModelChange)="filterHistory($event)" autocomplete="off"> {{ 'All' | translate }}
+      </label>
+      <label class="btn btn-secondary" for="filter_delegated" [class.active]="filter === 'delegated'">
+        <input type="radio" [ngModel]="filter" value="delegated" name="filter" id="filter_delegated" (ngModelChange)="filterHistory($event)" autocomplete="off"> {{ 'Delegated' | translate }}
+      </label>
+      <label class="btn btn-secondary" for="filter_undelegated" [class.active]="filter === 'undelegated'">
+        <input type="radio" [ngModel]="filter" value="undelegated" name="filter" id="filter_undelegated" (ngModelChange)="filterHistory($event)" autocomplete="off"> {{ 'Undelegated' | translate }}
+      </label>
+    </div>
+  </div>
+
   <div class="row">
-    <div *ngIf="history.length > 0;else noHistory" class="list-group w-100">
+    <div *ngIf="filteredHistory.length > 0;else noHistory" class="list-group w-100">
       <div *ngFor="let item of historyItems" class="list-group-item list-group-item-action list-group-item-{{item.color}}">
-        <a routerLink="/result/{{item.id}}"> {{ item.creation_time | amFromUtc | date:'yyyy-MM-dd HH:mm zzzz' }}</a>
-
+        <div class="d-flex w-100 justify-content-between">
+          <a routerLink="/result/{{item.id}}"> {{ item.creation_time | amFromUtc | date:'yyyy-MM-dd HH:mm zzzz' }}</a>
+          <small *ngIf="item.undelegated" class="text-muted">{{ "Undelegated" | translate }}</small>
```

Comment:
Would it be possible to use a singular form here for languages that support pluralization?

Reply:
> Would it be possible to use a singular form here for languages that support pluralization?

Can you explain what you mean?

Reply:
This `<small>Undelegated</small>` is displayed per test. In French, you would like to display `Non délégué` and not `Non délégués` since it applies to only one test. However the term is currently defined for the buttons that would be translated using a plural form in French `Non délégués`. So I was asking if it would be possible to distinct both cases, singular and plural forms.

Reply:
In Swedish it would be "odelegerade" (plur) and "odelegerad" (sing), respectively. Could we accept a compromise, "odelegerad(e)" and "non délégué(s)", respectively, too keep the number of translated strings down?

There is, however, a risk that some language has a plural inflection for that phrase that changes the stem too, such as "bok - böcker" (book, sing and plur).

I guess I agree that we should have both forms.